include/zephyr/net/prometheus/pm_collector.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2024 Mustafa Abdullah Kus, Sparse Technology
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ *
+ * @brief prometheus collector APIs.
+ */
+
+#ifndef ZEPHYR_INCLUDE_PM_COLLECTOR_H_
+#define ZEPHYR_INCLUDE_PM_COLLECTOR_H_
+
+#include <zephyr/net/prometheus/pm_metric.h>
+
+#include <stddef.h>
+
+typedef struct {
```

Comment:
We normally use typedefs only for function pointers in Zephyr and avoid using typedefs for structs. So with this in mind you could just have `struct pm_collector_node { ... }` here. Similar comment to all the other typedefs you have in the header files.

Also, all the structs and their fields need to be documented if they are considered to be "public" i.e., used by the application. For "private" structs, you should use `/** @cond INTERNAL_HIDDEN */` ... `/** @endcond */` to prevent the struct from showing in the generated documentation. See examples in other header files.